### PR TITLE
Add check for node version

### PR DESCRIPTION
**Pull request by imurchie**
Add a check to make sure the node version is correct. Also move the node checks (existence and version) into a general check, so that it is checked for both Android and iOS.

### CODE REVIEW

File: lib/factory.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+import { Doctor } from './doctor';
+import generalChecks from './general';
 import iosChecks from './ios';
 import androidChecks from './android';
 import devChecks from './dev';
 import demoChecks from './demo';
 
 
-let checks = { iosChecks, androidChecks, devChecks, demoChecks };
+let checks = { generalChecks, iosChecks, androidChecks, devChecks, demoChecks };
```
Comment (jlipps) [inline lib/factory.js]: _technically_ this shouldn't have spaces inside the curly braces

File: lib/general.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { ok, nok } from './utils';
+import { exec } from 'teen_process';
+import { DoctorCheck } from './doctor';
+import NodeDetector from './node-detector';
+
+
+let checks = [];
+
+// Node Binary
+class NodeBinaryCheck extends DoctorCheck {
```
Comment (jlipps) [inline lib/general.js]: seems odd to me that someone would be able to run this without the `node` binary being present. but maybe this is somehow possible given differing environments?

Comment (jlipps) [inline lib/general.js]: ah, I see that you're just moving this, n/m

Comment (imurchie) [inline lib/general.js]: Yes. Previously if you only ran Android check it wouldn't get run.

But yes, it seems like you would not be able to run the `appium-doctor` program at all if you didn't have Node installed in some way.

File: lib/factory.js (hunk 1 above)
Comment (imurchie) [inline lib/factory.js]: Fixed!

**Pull request merged**